src/Model/List/ArrayList.php

```diff
@@ -455,19 +455,24 @@ protected function parseSortColumn($column, $direction = null)
      */
     public function sort(...$args): static
     {
-        if (count($args ?? [])==0) {
+        $numArgs = count($args);
+        // If explicitly not sorting, just return the list back
+        if ($numArgs == 0) {
             return $this;
         }
-        if (count($args ?? [])>2) {
+        if ($numArgs === 1 && $args[0] === null) {
+            return $this;
+        }
```

Comment:
Setting sort to `null` means "don't perform sort operations".
Since `ArrayList` sorts at the time the method is called instead of at iteration time, passing in `null` should just do nothing.

Without this fix, we end up hitting `new InvalidArgumentException("Bad arguments passed to sort()")` which is:
1. inconsistent with other list implementations
2. problematic if a list which is usually a `DataList` has been swapped out with `ArrayList` in a place where typing isn't very strict

It was causing a test to fail because of 2 above.

Reply:
All other changes in this method are unrelated readability improvements